File: server/src/discord/commands/pasta.rs
use crate::discord::commands::shared::{Category, CommandData, CreateCommand, Interaction};
use crate::discord::helper::Helper;
use crate::discord::ids::PASTA;
use crate::error::Error;
use crate::{get_options, index_option};
use rocket::serde::json;
use serde::Deserialize;
use serenity::client::Context;

use super::shared::create_index_option;

pub const DATA: CommandData = CommandData {
    admin_only: false,
    name: "pasta",
    category: Category::Misc,
};

pub fn create(command: &mut CreateCommand) -> &mut CreateCommand {
    command
        .description("sends the ulina copy-pasta of your choice")
        .add_option({
            let mut option = create_index_option(PASTAS.iter(), PASTA, |pasta| pasta.name.clone());
            option.required(true);
            option
        })
}

lazy_static! {
    static ref PASTAS: Vec<Pasta> =
        json::from_str(include_str!("../../../config/pasta.json")).unwrap();
}

#[derive(Deserialize)]
struct Pasta {
    name: String,
    content: String,
}

pub async fn pasta(ctx: &Context, interaction: &Interaction) -> Result<(), Error> {
    let pasta = index_option!(interaction.data.options, PASTA, PASTAS, "pasta");

    interaction
        .message(&ctx.http, |data| data.content(&pasta.content))
        .await;

    Ok(())
}

File: UlinaSiteNode/bot/commands/pasta.ts
import {SlashCommandBuilder} from "@discordjs/builders";
import fs = require('fs');
import {CommandInteraction} from "discord.js";
import {UlinaCommand} from "../bot";

interface Pasta{
    name: string,
    content: string
}
const pastas: Pasta[] = JSON.parse(fs.readFileSync("./data/pasta.json", "utf-8")).pastas;

module.exports = {
    data: new SlashCommandBuilder()
        .setName('pasta')
        .setDescription('sends the ulina copy-pasta of your choice')
        .addIntegerOption(input => {
                const option = input.setName("pasta").setRequired(true)
                    .setDescription("the selected pasta");

                pastas.forEach((pasta, index) => {
                    option.addChoices({name: pasta.name, value: index})
                });
                return option;
            }
        ),
    async execute(interaction: CommandInteraction) {
        const pasta = pastas[interaction.options.getInteger("pasta")];
        await interaction.reply(pasta.content);
    },
    category: "Misc"
} as UlinaCommand;